src/objects/core/query.py

```diff
@@ -32,3 +30,17 @@ def filter_for_date(self, date):
 
     def filter_for_registration_date(self, date):
         return self.filter(records__registration_at__lte=date).distinct()
+
+
+class ObjectRecordQuerySet(models.QuerySet):
+    def filter_for_date(self, date):
+        return (
+            self.filter(start_at__lte=date)
+            .filter(models.Q(end_at__gte=date) | models.Q(end_at__isnull=True))
+            .order_by("-index")
+        )
+
+    def filter_for_registration_date(self, date):
+        return self.filter(registration_at__lte=date).order_by(
+            "-registration_at", "-index"
```

Comment:
The `order_by` is missing in `ObjectQuerySet` counterpart above?

Reply:
`ObjectRecordQuerySet.filter_for_registration_date` has sorting since we extract the first record later. `ObjectQuerySet.filter_for_registration_date` filters objects themselves not their records, and it is used in `ObjectViewSet`, where queryset is already sorted.